src/Components/Sidebar/Sidebar.js

```diff
@@ -14,12 +14,44 @@ export default function Sidebar() {
       navigate("/");
     });
   }, [navigate]);
+  const optionsDemo = [
+    { name: "weekly progress", id: "abc" },
+    { name: "cool talks", id: "def" },
+    { name: "In the middle of june", id: "ghi" },
+  ];
+  const [menuOptions, setMenuOptions] = useState(optionsDemo);
+  console.log(setMenuOptions);
+  // will set it once api starts working
+  const sidebarMenuOptions = menuOptions.map((item, key) => {
+    return (
+      <div
+        className="sidebar__menu--options"
+        key={key}
+        onClick={() => console.log(`backend/document/${item.id} is clicked`)}
+        // will convert it to another function and add useCallback
+      >
+        {item.name}
+      </div>
+    );
+  });
+
+  const SidebarMenu = useCallback(() => {
+    return <div>{sidebarMenuOptions}</div>;
+  }, [sidebarMenuOptions]);
 
   return (
     <div className="sidebar">
       <Link to="/" className="sidebar__logo">
         <img src={logo} alt="logo" />
       </Link>
+      <div className="sidebar__menu">
+        <div className="sidebar__menu--headers">
+          <p className="sidebar__menu--headers--title">Documents:</p>
+          <button className="sidebar__menu--headers--btn">+</button>
+          {/* will set onClick functionalities once api is done */}
+        </div>
+        <SidebarMenu />
```

Comment:
SideBarMenu is a function - should not be used as a component; can cause unspecified behaviour;

Reply:
if it's supposed to be a component, then don't use useCallback; 
useCallback is for optimizing functions - I'm not clear on how it affects components - if you found any documentation on that, I'd like to see

Reply:
No, it does on functions only, components inside are optimized.